src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -153,7 +153,9 @@ private void AddRuleset()
         /// </summary>
         private void TrySetNonConditionalRuleSet()
         {
-            var hasUserSpecifiedRuleSet = propertyInformationMap.Keys.Any(x => !IsDefaultMicrosoftRuleSet(x.Value as string));
+            var solutionRuleSetProvider = serviceProvider.GetService<ISolutionRuleSetsInformationProvider>();
+            var ruleSetsInfo = solutionRuleSetProvider.GetProjectRuleSetsDeclarations(initializedProject).ToArray();
+            var hasUserSpecifiedRuleSet = ruleSetsInfo.Any(x => !IsDefaultMicrosoftRuleSet(x.RuleSetPath));
```

Comment:
This doesn't feel quite right: there's quite an overlap between this processing and what happens in `CalculateRuleSetInformation`.
`CalculateRuleSetInformation` which has already fetched the `rulesetsInfo` so calculating the `hasUserSpecifiedRuleSet` there would be a one-liner.Also, the naming of `CalculateRuleSetInformation` suggests it's intended to be the place were the calculations are done.

Reply:
So you're suggesting to extract `hasUserSpecifiedRuleSet` to a member?

Reply:
Yep. Logically it's the same kind of data as `propertyInformationMap`.